docs/reference/search/search-your-data/highlighting.asciidoc

```diff
@@ -171,9 +171,6 @@ NOTE: Only text, match_only_text, and keyword fields are highlighted when
 you use wildcards. If you use a custom mapper and want to highlight on a
 field anyway, you must explicitly specify that field name.
 
-force_source:: Highlight based on the source even if the field is
-stored separately. Defaults to `false`.
```

Comment:
One thing that caught my eye is also: if this was an internal parameter that only the percolator needed, why was the parameter exposed at REST? I have some doubts about making it a no-op straight-away, because there are potential existing users, or do we know for a fact that nobody relies on it?

Reply:
`force_source` predates me. It existed when I started looking at highlighters like 9 years ago and I always wondered why you'd want to set it. In generally I never `store`d a field because I always wanted the whole source for a small number of docs.

Reply:
The type of thing having telemetry for would help, hopefully it would indicate `0` usages.

Reply:
So I did a bit of git archaeology: the parameter was added in 10e2528cceb404165822862602eb1326ccb1fba5 back in 2013, before we had separate XContent and transport protocols.  So if you wanted to add an internal parameter, it had to go in XContent and was therefore visible to the outside world.  I am reasonably confident that this does in fact have `0` usages, and that if it doesn't, the people using it will not notice if it becomes a no-op as currently it will only have a negative impact on performance.

Reply:
thanks for taking the time to look into it and explaining.